src/libsync/owncloudpropagator.cpp

```diff
@@ -578,16 +578,16 @@ PropagatorJob::JobParallelism PropagateDirectory::parallelism()
 {
     // If any of the non-finished sub jobs is not parallel, we have to wait
 
-    // FIXME!  we should probably cache this result
-
     if (_firstJob && _firstJob->_state != Finished) {
         if (_firstJob->parallelism() != FullParallelism)
             return WaitForFinished;
     }
 
-    // FIXME: use the cached value of finished job
-    for (int i = 0; i < _subJobs.count(); ++i) {
-        if (_subJobs.at(i)->_state != Finished && _subJobs.at(i)->parallelism() != FullParallelism) {
+    QListIterator<PropagatorJob *> subJobsIterator(_subJobs);
+
+    while (subJobsIterator.hasNext()) {
+        subJobsIterator.next();
+        if (subJobsIterator.peekPrevious()->_state != Finished && subJobsIterator.peekPrevious()->parallelism() != FullParallelism) {
             return WaitForFinished;
```

Comment:
minor: Would prefer `const auto job = subJobsIterator.next();` and then use `job` over `subJobsIterator.peekPrevious()`. Usage of `peekPrevious()` makes me think that something special is happening. Same in the other loop below.

I'd even be fine calling `subJobsIterator` `it`.